packages/core/src/components/graphs/meter.ts

```diff
@@ -0,0 +1,90 @@
+// Internal Imports
+import { Component } from "../component";
+import { DOMUtils } from "../../services";
+import { Tools } from "../../tools";
+
+// D3 Imports
+import { scaleLinear } from "d3-scale";
+import { Roles } from "../../interfaces";
+
+export class Meter extends Component {
+	type = "meter";
+
+	render(animate = true) {
+		const self = this;
+		const svg = this.getContainerSVG();
+		const options = this.model.getOptions();
+		const data = this.model.getDisplayData();
+		const status = this.model.getStatus();
+
+		const { width } = DOMUtils.getSVGElementSize(this.parent, { useAttrs: true });
+		const { groupMapsTo } = options.data;
+
+		// each meter has a scale for the value but no visual axis
+		const xScale = scaleLinear()
+			.domain([0, 100])
+			.range([0, width]);
+
+		// draw the container to hold the value
+		DOMUtils.appendOrSelect(svg, "rect.container")
+			.attr("x", 0 )
+			.attr("y", 0 )
+			.attr("width", width)
+			.attr("height", Tools.getProperty(options, "meter", "height"));
+
+		// value larger than 100 will display as 100% on meter chart
+		const dataset = data.value <= 100 ? data : data["value"] = 100;
+
+		// rect with the value binded
+		const value = svg.selectAll("rect.value")
+			.data([dataset]);
+
+		// if user provided a color for the bar, we dont want to attach a status class
+		const userProvidedScale = Tools.getProperty(options, "color", "scale");
+
+		// draw the value bar
+		value.enter()
+			.append("rect")
+			.classed("value", true)
+			.merge(value)
+			.attr("x", 0 )
+			.attr("y", 0 )
+			.attr("height", Tools.getProperty(options, "meter", "height"))
+			.classed(`status--${status}`, status != null && !userProvidedScale)
+			.transition(this.services.transitions.getTransition("meter-bar-update", animate))
+			.attr("width", d => xScale(d.value))
+			.attr("fill", d => self.model.getFillColor(d[groupMapsTo]))
+			// a11y
+			.attr("role", Roles.GRAPHICS_SYMBOL)
+			.attr("aria-roledescription", "value")
+			.attr("aria-label", d => d.value);
+
+		// draw the peak
+		const peakValue = Tools.getProperty(options, "meter", "peak") ?? null;
```

Comment:
what's does this do? `Tools.getProperty(options, "meter", "peak") ?? null`

Reply:
if there is a peak supplied, it is assigned to the `peakValue`, otherwise if it is undefined or null it will be default assigned to `null`(and chart wont render a peak)

Reply:
`Tools.getProperty` will already return `null` if it fails anywhere in its process